CMakeLists.txt

```diff
@@ -47,7 +36,19 @@ else()
 endif()
 
 ## Add serial library
+set(serial_HEADERS
+    include/serial/serial.h
+    include/serial/v8stdint.h
+)
```

Comment:
```suggestion
```
You don't need to specify headers as source files if you don't want to.

Reply:
```
cmake: Add defined so-version and public header to lib.
Adding a so-version means defining an ABI level. This level is decoupled
from the ordinary version, even a major version change doesn't
necessarily mean that the so-version should change (and thus have all
dependencies to be rebuilt).

Adding the public header to clarify the setup.

Gbp-Pq: Name 0002-cmake-Add-defined-so-version-and-public-header-to-li.patch
```
This was added here https://github.com/wjwwood/serial/pull/231/commits/ed0e389352359123ad400cc785b76158ac303933 which I cherry-picked https://github.com/wjwwood/serial/pull/285/commits/61da1e2c994ebf3985521d9b6b431d6171c9983b

Reply:
_v8stdint.h_ is actually problematic. It is now  long time since I worked with this, but IIRC I **think** the situation is:

  - _v8stdint.h_ is basically a band-aid for systems not having _stdint.h_ available, notably windows.
  - The definitions in _v8stdint.h_ clashes with other files on some systems.

That is, the correct solution would be to use cmake to probe for `stdint.h` and only include `v8stdint.h` if it's required.  Needs to be double-checked, though. If it is indeed required anyway, cmake should probe for _v8stdint.h_ and only include it if it's not available.

On Fedora, _v8stdint.h_ is part of the v8 package.

Reply:
Of course, this is not only about including v8stdint.h or not in the package. It is also about conditionalizing the source files so they include the correct file, _stdint.h_ or _v8stdint.h_. I made a quick hack for this in the Debian package, but it probably needs polish before being merged into the upstream package here.

Reply:
> You don't need to specify headers as source files if you don't want to.

This is IMHO the wrong way to go. Recommended cmake best practice is certainly to include all used files in the project.

Reply:
> Recommended cmake best practice is certainly to include all used files in the project.

Do you have a source for this? It's not something I've ever heard. The main reason for doing it is to satisfy Visual Studio users who want to see headers in the Solution Explorer.

Reply:
I have seen a numbner of cmake projects, none of without  the headers in the source lists. In this case, I think the you need to come up with some trustworthy source recommending not to include the headers.

You have already started the list of functionality which breaks without a complete list of source files, including headers.  I can assure you there is more, and that we currently not use it is not much of an argument.

BTW: how would you install headers not being part of the sources?

Reply:
> I think that you need to come up with some trustworthy source recommending not to include the headers.

I'd argue the burden of proof is on the person who wants to write more code.

> You have already started the list of functionality which breaks without a complete list of source files

Nothing breaks for Visual Studio users. It's merely an inconvenience.

> how would you install headers not being part of the sources?

Same way as always (prior to `FILE_SET`s). You install the `include` directory or you manually glob all the headers and install them. Both accomplish the same thing in most cases. Adding headers as source files does not affect installation.

Reply:
I simply don't have time for this, sorry.  It's up to the maintainer to judge.

Reply:
My experience has been that adding headers is not required at all. It does help some editors who want to list "files used in targets" without involving a compiler, but that's all. Whether or not it's recommended by CMake, I'm not sure about.

That being said, I don't care if they are included in this way. If it helps someone and they contribute it then I think it's fine.

Reply:
Fair enough. But, as you said, cstdint.h is the better alternative to v8stdint.h making this to be serial.h only